docs/outlook/mobile-log-appointments.md

```diff
@@ -0,0 +1,553 @@
+    }
+
+    function getGlobal() {
+      return typeof self !== "undefined"
+        ? self
+        : typeof window !== "undefined"
+        ? window
+        : typeof global !== "undefined"
+        ? global
+        : undefined;
+    }
+
+    const g = getGlobal();
+    g.logCRMEvent = logCRMEvent;
+    ```
+
+### Implement viewing appointment notes
+
+The **Log** button can be toggled to display **View** by setting the **EventLogged** custom property reserved for this purpose. Then when the user the user selects the **View** button, they can look at their logged notes for this appointment.
+
+Add the following function to **./src/commands/commands.js**. This function sets the **EventLogged** custom property on the current appointment item.
+
+```js
+function updateCustomProperties(event) {
+  Office.context.mailbox.item.loadCustomPropertiesAsync(
+    function callback(customPropertiesResult) {
+      if (customPropertiesResult.status === Office.AsyncResultStatus.Succeeded) {
+        let customProperties = customPropertiesResult.value;
+        customProperties.set("EventLogged", true);
+        customProperties.saveAsync(
+          function callback(setSaveAsyncResult) {
+            if (setSaveAsyncResult.status === Office.AsyncResultStatus.Succeeded) {
+              console.log("EventLogged custom property saved successfully.");
+            }
+          }
+        );
+      }
+    }
+  );
+}
+```
+
+You can then call it after the add-in successfully logs the appointment notes. For example, you can call it from **LogCRMEvent** as follows.
+
+```js
+function logCRMEvent(event) {
+  console.log(`Subject: ${Office.context.mailbox.item.subject}`);
+  Office.context.mailbox.item.body.getAsync(
+    "html",
+    { asyncContext: "This is passed to the callback" },
+    function callback(result) {
+      if (result.status === Office.AsyncResultStatus.Succeeded) {
+        // Replace `event.completed({ allowEvent: true });` with the following statement.
+        updateCustomProperties(event);
+      } else {
+        console.error("Failed to get body.");
+        event.completed({ allowEvent: false });
+      }
+    }
+  );
+}
+```
+
+Your add-in defines the log viewing experience. For example, you can display the logged appointment notes in a dialog when the user selects the **View** button. For details on using dialogs, refer to [Use the Office dialog API in your Office Add-ins](../develop/dialog-api-in-office-add-ins.md).
+
+### Implement deleting the appointment log
+
+If you'd like to enable your users to undo logging or delete the logged appointment notes so a replacement log can be saved, use Microsoft Graph to [clear the custom properties object](/graph/api/resources/extended-properties-overview?view=graph-rest-1.0&preserve-view=true) when the user selects the appropriate button in the ribbon.
+
+Add the following function to **./src/commands/commands.js** to clear the **EventLogged** custom property on the current appointment item.
```

Comment:
Should we mention that the developer should use either of graph API approach or from an addin session using the below code